exporter/elasticsearchexporter/data_stream_router.go

```diff
@@ -6,12 +6,39 @@ package elasticsearchexporter // import "github.com/open-telemetry/opentelemetry
 import (
 	"fmt"
 	"regexp"
+	"strings"
+	"unicode"
 
 	"go.opentelemetry.io/collector/pdata/pcommon"
 )
 
 var receiverRegex = regexp.MustCompile(`/receiver/(\w*receiver)`)
 
+const (
+	maxDataStreamBytes       = 100
+	disallowedNamespaceRunes = "\\/*?\"<>| ,#:"
+	disallowedDatasetRunes   = "-\\/*?\"<>| ,#:"
+)
+
+// Sanitize the datastream fields (dataset, namespace) to apply restrictions
+// as outlined in https://www.elastic.co/guide/en/ecs/current/ecs-data_stream.html
```

Comment:
If I understand, it looks like datastream names are required to follow [index naming restrictions](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-create-index.html#indices-create-api-path-params), which introduces more limits as well. Should we add those requirements as well, or are they irrelevant? (We can add more restrictions later, just bringing this up to ensure we're not missing anything).

Reply:
Hi @crobert-1. This is a "tricky" one. Thanks for the comment.

The datastream name is composed of `{data_stream.type}-{data_stream.dataset}-{data_stream.namespace}`. Here the `data_stream.type` restrictions are derived from the ES index restrictions, see [Restrictions on Values](https://github.com/elastic/ecs/blob/main/rfcs/text/0009-data_stream-fields.md)

Moreover, a lot of the index name restrictions are implicit in `data_stream.type`. For example, the `logs`, `traces`, and `metrics` prefixes are hardcoded into `data_stream.type`, making it impossible for the name to start with `.` or `..`.

Reply:
That makes sense, I appreciate the clarification!